reporter-sqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,203 @@
+    public abstract Builder credentialsProvider(AWSCredentialsProvider credentialsProvider);
+    public abstract Builder clientConfiguration(ClientConfiguration clientConfiguration);
+    public abstract Builder performCheck(boolean performCheck);
+
+    public abstract Builder encoding(Encoding encoding);
+
+    /**
+     * Maximum size of a message. SQS max message size is 256KB including attributes.
+     * Default 256 less 30 bytes for message keys.
+     *
+     * @param messageMaxBytes
+     * @return
+     */
+    public abstract Builder messageMaxBytes(int messageMaxBytes);
+
+    public final AwsBufferedSqsSender build() {
+      return autoBuild();
+    }
+
+    abstract AwsBufferedSqsSender autoBuild();
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder(this);
+  }
+
+  abstract String queueUrl();
+  abstract boolean compressionEnabled();
+  abstract boolean performCheck();
+  @Nullable abstract ExecutorService dispatchExecutor();
+  @Nullable abstract AWSCredentialsProvider credentialsProvider();
+  @Nullable abstract ClientConfiguration clientConfiguration();
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
+    if (performCheck()) {
+      try {
+        get().getQueueAttributes(queueUrl(), Collections.singletonList("QueueArn"));
+        return CheckResult.OK;
+      } catch (Throwable e) {
+        return CheckResult.failed(new Exception(e));
+      }
+    }
+
+    return CheckResult.OK;
+  }
+
+  @Override protected AmazonSQSBufferedAsyncClient compute() {
+    AWSCredentialsProvider creds = (credentialsProvider() != null) ?
+        credentialsProvider() : new DefaultAWSCredentialsProviderChain();
+
+    AmazonSQSAsyncClient client;
+    if (clientConfiguration() != null && dispatchExecutor() != null) {
+      client = new AmazonSQSAsyncClient(creds, clientConfiguration(), dispatchExecutor());
+    } else if (clientConfiguration() != null && dispatchExecutor() == null) {
+      client = new AmazonSQSAsyncClient(creds, clientConfiguration());
+    } else if (clientConfiguration() == null && dispatchExecutor() != null) {
+      client = new AmazonSQSAsyncClient(creds, dispatchExecutor());
+    } else {
+      client = new AmazonSQSAsyncClient(creds);
+    }
+
+    return new AmazonSQSBufferedAsyncClient(client);
+  }
+
+  @Override public int messageSizeInBytes(List<byte[]> encodedSpans) {
+    return encoding().listSizeInBytes(encodedSpans);
+  }
+
+  @Override public void sendSpans(List<byte[]> list, Callback callback) {
+    if (closeCalled.get()) throw new IllegalStateException("closed");
+
+    checkNotNull(list, "list of encoded spans must not be null");
+
+    byte[] encodedSpans = BytesMessageEncoder.forEncoding(encoding()).encode(list);
+
+    get().sendMessageAsync(new SendMessageRequest(queueUrl(), "z")
```

Comment:
prefer "zipkin" please :) you can always configure it to the more hip and less bytes 'z'. It is easier to not have to explain unnecessary differences between transport defaults.